web/src/ui/components/RequestBuilder.js

```diff
@@ -0,0 +1,114 @@
+import { Select, JsonTextarea } from "../commons"
+import { isValidJson, getStringifiedJSON } from "../../util"
+import { useState } from "react"
+
+/*
+	TODO
+	JSON Textarea formatierungen
+	Path Logik auf eigene Komponente
+
+*/
+
+const RequestBuilder = () => {
+    const [method, setMethod] = useState("")
+    const [path, setPath] = useState("")
+    const [header, setHeader] = useState({})
+    const [body, setBody] = useState({})
+    const [bodyDisabled, setBodyDisabled] = useState(true)
+    var headerInputTimeout, bodyInputTimeout
+
+    const handleMethodChange = (selectedMethod) => {
+        setMethod(selectedMethod)
+        enableOptions(selectedMethod)
+    }
+
+    const removeCurrentOptions = () => {
+        setBodyDisabled(true)
+        //...
+    }
+
+    const enableOptions = (selectedMethod) => {
+        removeCurrentOptions()
+        if (selectedMethod.toLowerCase() === "post") {
+            setBodyDisabled(false)
+        }
+        //...
+    }
+
+    const handlePathChange = (event) => {
+        setPath(event.target.value)
+    }
+
+    const handleHeaderChange = (header) => {
+        clearTimeout(headerInputTimeout)
+        headerInputTimeout = setTimeout(() => {
+            if ((header = isValidJson(header))) {
+                setHeader(header)
+            } else {
+                //evtl Error bars triggern etc
+            }
+        }, 2000)
+    }
+    const handleBodyChange = (body) => {
+        clearTimeout(bodyInputTimeout)
+        bodyInputTimeout = setTimeout(() => {
+            if ((body = isValidJson(body))) {
+                setBody(body)
+            } else {
+                //evtl Error bars triggern etc
+            }
+        }, 2000)
+    }
+
+    const handleSubmit = () => {
+        const requestParams = {
+            method,
+            path,
+            header,
+            body
+        }
+        console.log("Submitting request:", requestParams)
+        // Hier kannst du deine Anfrage senden, z.B. mit fetch oder axios
+    }
+
+    const selectProps = {
+        options: ["POST", "GET", "DELETE", "PUT", "HEAD"],
+        label: "method",
+        className:
+            "bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500",
```

Comment:
ah okay, so hast du das verstanden, was ich geschrieben hatte. Ich meinte das eigentlich so, dass die Form-Komponente schon ihren eigenes styling macht (damit es überall gleich aussieht) aber das man zusätzlich über className noch weitere Klassen injizieren kann...also in der Komponente mache ich dann sowas wie:
```
function myComponent({ className, disabled = false, .... }) {
    // standard-styling...
    const cls = [""bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500"]
    
    // injizierte Klassen
    if (className) cls.push(className)
     
     // feature-abhängige Klassen
     if (disabled) cls.push('transparent')
     ...
     return (
        <div className={cls.join(' ')} ...>
     )
}
```
Wie ich aber geschrieben hatte, bin ich mir noch nicht sicher, ob wir das auch hier so machen sollen (wegen tailwind)